src/Microsoft.Diagnostics.Runtime/src/Extensions/IAddressableTypedEntityExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.Diagnostics.Runtime.Extensions
+{
+    public static class IAddressableTypedEntityExtensions
+    {
+        /// <summary>
+        /// Gets the field value from <see cref="IAddressableTypedEntity"/> with respect to field nature (either reference, or value type).
+        /// </summary>
+        /// <param name="entity">The entity to read field value from.</param>
+        /// <param name="fieldName">The name of the field to get value.</param>
+        /// <exception cref="ArgumentNullException">if entity has no type.</exception>
+        /// <exception cref="ArgumentException">if field with matching name was not found.</exception>
+        /// <returns></returns>
+        public static IAddressableTypedEntity GetFieldFrom(this IAddressableTypedEntity entity, string fieldName)
```

Comment:
Please replace all instances of `var` with a concrete type in ClrMD itself.  (I don't care about the test projects.)

It makes the code a LOT more readable to show what type we are dealing with.